javapoet/src/main/java/com/palantir/javapoet/TypeSpec.java

```diff
@@ -263,101 +272,114 @@ public Builder toBuilder() {
-                            codeWriter.emit(",");
-                        }
-                        codeWriter.emit(" $T", type);
-                        firstType = false;
+            List<TypeName> extendsTypes;
+            List<TypeName> implementsTypes;
+            if (kind == Kind.INTERFACE) {
+                extendsTypes = superinterfaces;
+                implementsTypes = Collections.emptyList();
+            } else {
+                extendsTypes = superclass.equals(ClassName.OBJECT)
+                        ? Collections.emptyList()
+                        : Collections.singletonList(superclass);
+                implementsTypes = superinterfaces;
+            }
+
+            if (!extendsTypes.isEmpty()) {
+                codeWriter.emit(" extends");
+                boolean firstType = true;
+                for (TypeName type : extendsTypes) {
+                    if (!firstType) {
+                        codeWriter.emit(",");
                     }
+                    codeWriter.emit(" $T", type);
+                    firstType = false;
                 }
+            }
 
-                if (!implementsTypes.isEmpty()) {
-                    codeWriter.emit(" implements");
-                    boolean firstType = true;
-                    for (TypeName type : implementsTypes) {
-                        if (!firstType) {
-                            codeWriter.emit(",");
-                        }
-                        codeWriter.emit(" $T", type);
-                        firstType = false;
+            if (!implementsTypes.isEmpty()) {
+                codeWriter.emit(" implements");
+                boolean firstType = true;
+                for (TypeName type : implementsTypes) {
+                    if (!firstType) {
+                        codeWriter.emit(",");
                     }
+                    codeWriter.emit(" $T", type);
+                    firstType = false;
                 }
+            }
 
-                if (!permittedSubclasses.isEmpty()) {
-                    codeWriter.emit(" permits");
-                    boolean firstType = true;
-                    for (TypeName type : permittedSubclasses) {
-                        if (!firstType) {
-                            codeWriter.emit(",");
-                        }
-                        codeWriter.emit(" $T", type);
-                        firstType = false;
+            if (!permittedSubclasses.isEmpty()) {
+                codeWriter.emit(" permits");
+                boolean firstType = true;
+                for (TypeName type : permittedSubclasses) {
+                    if (!firstType) {
+                        codeWriter.emit(",");
                     }
+                    codeWriter.emit(" $T", type);
+                    firstType = false;
                 }
+            }
 
-                codeWriter.popType();
+            codeWriter.popType();
+            codeWriter.emit(" {\n");
```

Comment:
nit: Keep the existing whitespace

```suggestion

            codeWriter.emit(" {\n");
```